fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -325,21 +377,27 @@ private NamedParameterJdbcTemplate getNamedParameterJdbcTemplate() {
 
     @Override
     public Stream<String> getContains(final String txId, final FedoraId fedoraId) {
-        final String resourceId = fedoraId.getFullId();
+        final String resourceId = fedoraId.isMemento() ? fedoraId.getBaseId() : fedoraId.getFullId();
```

Comment:
Would `getResouceId()` not do what you want here?

Reply:
`getResourceId()` handles ACLs and NonRdfSourceDescriptions, but not Mementos.

Reply:
But, the output of `getResourceId` can never contain a memento because it constructs an id based on the `baseId`.

Reply:
I understand, I can change this.